src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/usersettings/changesignature/ChangeSignatureService.java

```diff
@@ -0,0 +1,38 @@
+package de.dhbw.karlsruhe.students.mailflow.core.application.usersettings.changesignature;
+
+import de.dhbw.karlsruhe.students.mailflow.core.application.auth.AuthSessionUseCase;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.auth.LoadingUsersException;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.user.UserSettings;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.user.UserSettingsRepository;
+import de.dhbw.karlsruhe.students.mailflow.external.infrastructure.authorization.RemoveSettingsException;
+import de.dhbw.karlsruhe.students.mailflow.external.infrastructure.authorization.SaveSettingsException;
+
+public class ChangeSignatureService implements ChangeSignatureUseCase {
+
+  private final UserSettingsRepository userSettingsRepository;
+  private final AuthSessionUseCase authSession;
+
+  public ChangeSignatureService(
+      UserSettingsRepository userSettingsRepository, AuthSessionUseCase authSession) {
+    this.userSettingsRepository = userSettingsRepository;
+    this.authSession = authSession;
+  }
+
+  @Override
+  public void updateSignature(String newSignature)
+      throws LoadSettingsException, SaveSettingsException {
+
+    UserSettings userSettings = new SignatureBuilder().withSignature(newSignature).withAddress(authSession.getSessionUserAddress()).build();
+    userSettingsRepository.updateUserSettings(userSettings);
+  }
+
+  @Override
+  public void resetSignature() throws LoadSettingsException, SaveSettingsException {
+    userSettingsRepository.removeUserSettings(authSession.getSessionUserAddress());
```

Comment:
this should just empty out the signature field, not delete all the settings.

In the future, we might have more settings, e.g. e-mail sorting rules or the like